### PR TITLE
Fire calypso_user_registration_complete on new user registration

### LINKED ISSUE #7826: Conversion: add tracking event for new user creation
From what I can see there is no tracks event that gets fired when a new user is created in the signup flow. There are some tracks events for which we can infer that a new user has been created, but it's not obvious. I believe this is `calypso_signup_step_complete` where `step_name === 'user-survey'`.

There's `calypso_signup_complete`, but that gets fired whether this is a new user or not. I think we need a `calypso_user_registration_complete` event that we can use to better understand our funnels.

### PR DESCRIPTION
**Pull request by lamosty**
Fixes #7826. Adds a new Tracks/Google Analytics event which is fired after a new user registers. No additional data is sent with the event.

## Testing Instructions

1. Navigate to calypso.localhost:3000/start in an incognito window;
2. Turn on analytics JS debugging by running `localStorage.setItem('debug', 'calypso:analytics');` in your browser's dev console followed by full page reload;
3. Go through the signup flow;
4. After completing the new user signup form, look at your dev console for `calypso_user_registration_complete` event logging;
5. When reviewing the code, verify that the event is fired only after creating a brand new user account and only after it has been successfully created.

/cc @rralian 

Test live: https://calypso.live/?branch=add/new-user-creation-tracking-event

### CODE REVIEW

File: client/lib/signup/actions.js [hunk 1]
```diff
@@ -53,7 +53,15 @@ const SignupActions = {
 	},
 
 	processedSignupStep( step, errors, providedDependencies ) {
-		analytics.tracks.recordEvent( 'calypso_signup_actions_complete_step', { step: step.stepName } );
+		const { stepName } = step;
+
+		analytics.tracks.recordEvent( 'calypso_signup_actions_complete_step', { step: stepName } );
+
+		if ( stepName === 'survey-user' ) {
```
Comment (bisko) [inline client/lib/signup/actions.js]: We should avoid hardcoding step names wherever possible in signup. The steps can change at any moment and this code would stop working. 

For example there are currently two steps that are responsible for creating users - `user` and `survey-user`. The `user` step is used in some of the developer-only flows and will not fire the event.

I think the best way to do that is to add the code directly in [`step-actions.js`/`createAccount`](https://github.com/Automattic/wp-calypso/blob/1ae3209cb1169686f065087aa6a800afdc4bf9c0/client/lib/signup/step-actions.js#L181) where the code that handles the user creation request sits.

Comment (lamosty) [inline client/lib/signup/actions.js]: Thanks for the suggestion, I didn't know we have these `stepActions`. the `signup/` code is huge. :)

**Comment by gwwar** [comment]
@rralian did we still need this one?

File: client/lib/signup/step-actions.js [hunk 2]
```diff
@@ -179,6 +180,12 @@ module.exports = {
 			var errors = error && error.error ? [ { error: error.error, message: error.message } ] : undefined,
 				bearerToken = error && error.error ? {} : { bearer_token: response.bearer_token };
 
+			if ( ! errors ) {
+				// Fire after a new user registers.
+				analytics.tracks.recordEvent( 'calypso_user_registration_complete' );
```
Comment (artpi) [inline client/lib/signup/step-actions.js]: Are we able to pass here some props? Like site_URL, user ID, purchase?
It would be cool to have this data for slice&dice in analytics

**Comment by artpi** [comment]
> did we still need this one?

I think that WHEN will we need it, it will be too late. This event would be very useful for future benchmark funnels. I see no harm in introducing it

It would be nice to pass the purchased plan id to be able to calculate conversion in GA
( that seems like a promising option: https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#eventValue )
, other then that, it seems fine and ready to :ship:

**Comment by lancewillett** [comment]
@lamosty Ready to merge?

File: client/lib/signup/step-actions.js (hunk 2 above)
Comment (lamosty) [inline client/lib/signup/step-actions.js]: I tried to get these props but didn't find a way. We can always do that in another PR, though. :)

**Comment by lamosty** [comment]
Thanks for the review @artpi!
